### PR TITLE
[1713] Allow courses to be changed from confirmation components

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/eMZwfzbp/1713-s-publish-courses-allow-courses-to-be-changed-from-confirmation-component

### Changes proposed in this pull request

- Allow courses to be changed from confirmation components in both draft and non draft workflows
- Handle stashing/undoing behaviour

### CODE REVIEW

**Comment by JonF1** [comment]
[[S] Publish courses: allow courses to be changed from confirmation component](https://trello.com/c/eMZwfzbp/1713-s-publish-courses-allow-courses-to-be-changed-from-confirmation-component)

File: spec/components/trainees/confirmation/course_details/view_spec.rb [hunk 1]
```diff
@@ -24,42 +25,64 @@ module CourseDetails
...
+              expect(Course).to receive(:find_by).with(code: trainee.course_code).and_return(course)
+              render_inline(View.new(data_model: trainee))
+            end
 
-          it "renders the course type" do
-            expect(component.find(".govuk-summary-list__row.type-of-course .govuk-summary-list__value"))
-              .to have_text(trainee.training_route.humanize)
-          end
+            it "renders the course details" do
+              expect(component.find(".govuk-summary-list__row.course-details .govuk-summary-list__value"))
```
Comment (darokel) [inline spec/components/trainees/confirmation/course_details/view_spec.rb:50]: Could we put `component.find(".govuk-summary-list__row.course-details .govuk-summary-list__value"` into a `subject { component.find(".govuk-summary-list__row.course-details .govuk-summary-list__value" }` so we can assert on the subject? Then we won't need to duplicate the find across the other scenarios

File: app/forms/course_details_form.rb [hunk 2]
```diff
@@ -72,6 +78,7 @@ def other_age_range?
 
   def update_trainee_attributes
     trainee.assign_attributes({
+      course_code: nil,
```
Comment (archferns) [inline app/forms/course_details_form.rb:81]: is this always supposed to be nil?

Comment (darokel) [reply app/forms/course_details_form.rb:81]: this resets the course code value if a course has been entered manually via this form

**Review by archferns — APPROVED** [review]

**Pull request merged**